met-api/src/met_api/services/email_verification_service.py

```diff
@@ -131,12 +131,14 @@ def validate_object(email_verification: EmailVerificationSchema):
         if not is_active:
             raise ValueError('Email verification already verified')
 
-        verification_created_datetime = datetime.strptime(
-            email_verification.get('created_date'), EmailVerificationService.datetime_format)
-        verification_expiry_datetime = verification_created_datetime + \
-            timedelta(hours=EmailVerificationService.verification_expiry_hours)
-        if datetime.now() >= verification_expiry_datetime:
-            raise ValueError('Email verification is expired')
+        if email_verification.get('submission_id') is None:
```

Comment:
We should likely come up with a more permanent solution for expiring email verification tokens (likely a new expiration date column) but it has a few tricky things about it so I'm going with the simple fix for now

Reply:
afaik , we shouldnt expire the email token till the submission is open rite?then we can check the expiry agianst the engagement schedule date?

Reply:
thats right, it does check the engagement dates right below, this change will bypass the check for the 24 h expiration date if the token was generated for editing a submission

Reply:
i didn't want to have a specific expiration date set for the engagement dates since we still allow the engagement dates to be changed while its active